ingestion/src/metadata/profiler/metrics/static/unique_count.py

```diff
@@ -65,16 +65,12 @@ def query(
         only_once_cte = only_once.cte("only_once")
         return session.query(func.count().label(self.name())).select_from(only_once_cte)
 
-    def df_fn(self, df=None):
+    def df_fn(self, dfs=None):
         """
         Build the Unique Count metric
         """
-        from pandas import DataFrame  # pylint: disable=import-outside-toplevel
-
-        df = cast(DataFrame, df)
-
         try:
-            return df[self.col.name].nunique()
+            return sum(len(df[self.col.name].dropna().unique()) for df in dfs)
```

Comment:
Here we need to compute the count of values that are unique (occure only once) across all the dataframes.

we'll need 2 things:
1. keep a tally of value count
2. add to the tally count when we visit each dataframe

we'll need to do something like that (please check the implementation in details, don't just copy/past)

```python
from collections impot Counter

c = Counter()

for df in dfs:
    c.update(df[self.col.name].value_counts().to_dict(()

unique_count = len([k for k,v in c.items() if c == 1])
...
```

Reply:
I think the unique count needs to be calculated with the whole and not the chunks, correct me if i am wrong..because if we are checking with chunks, we might miss some scenarios where the data might be similar in different chunks